test/controllers/tradingNames/AddTradingNameControllerSpec.scala

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.tradingNames
+
+import base.SpecBase
+import forms.tradingNames.AddTradingNameFormProvider
+import models.{Index, TradingName}
+import org.mockito.ArgumentMatchers.{any, eq => eqTo}
+import org.mockito.Mockito.{times, verify, when}
+import org.scalatestplus.mockito.MockitoSugar
+import pages.tradingNames.{AddTradingNamePage, TradingNamePage}
+import pages.{EmptyWaypoints, JourneyRecoveryPage, Waypoints}
+import play.api.data.Form
+import play.api.inject.bind
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import repositories.AuthenticatedUserAnswersRepository
+import utils.FutureSyntax.FutureOps
+import viewmodels.checkAnswers.TradingNameSummary
+import views.html.tradingNames.AddTradingNameView
+
+class AddTradingNameControllerSpec extends SpecBase with MockitoSugar {
+
+  private val waypoints: Waypoints = EmptyWaypoints
+  private val index: Index = Index(0)
+
+  val formProvider = new AddTradingNameFormProvider()
+  val form: Form[Boolean] = formProvider()
+  private val baseAnswers = basicUserAnswersWithVatInfo.set(TradingNamePage(Index(0)), TradingName("foo")).success.value
+
+  lazy val addTradingNameRoute: String = routes.AddTradingNameController.onPageLoad(waypoints).url
+
+  "AddTradingName Controller" - {
+
+    "must return OK and the correct view for a GET" in {
+
+      val application = applicationBuilder(userAnswers = Some(baseAnswers)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, addTradingNameRoute)
+
+        val result = route(application, request).value
+
+        val view = application.injector.instanceOf[AddTradingNameView]
+
+        val list = TradingNameSummary.addToListRows(baseAnswers, waypoints, AddTradingNamePage())
+
```

Comment:
Probably worth doing a test just below the upper boundary as we are saying empty allows adding, 10 does not, so at some point between empty and 10 it switches to not

```scala
    "must allow adding a trading name when just below the maximum number of trading names" in {
      val userAnswers = (0 to 8).foldLeft(basicUserAnswersWithVatInfo) { case (a: UserAnswers, index: Int) =>
        a.set(TradingNamePage(Index(index)), TradingName("foo")).success.value
      }

      val application = applicationBuilder(userAnswers = Some(userAnswers)).build()

      running(application) {
        val request = FakeRequest(GET, addTradingNameRoute)

        val view = application.injector.instanceOf[AddTradingNameView]

        val list = TradingNameSummary.addToListRows(userAnswers, waypoints, AddTradingNamePage())

        val result = route(application, request).value

        status(result) mustBe OK
        contentAsString(result) mustBe view(form, waypoints, list, canAddTradingNames = true)(request, messages(application)).toString
      }
    }


```